### ISSUE TITLE
Output of just-diff not compatible with just-diff-apply for arrays

### ISSUE THREAD

**Issue by kbhokray** [angus-c/just#143]
When the output of just-diff on an array is run through just-diff-apply, the resulting array is not the same:

```
> let a = [1, 2, 3, 4, 5]
[ 1, 2, 3, 4, 5 ]
> let b = [5]
[ 5 ]
> let c = diff(a, b)
[ { op: 'remove', path: [ 1 ] },
  { op: 'remove', path: [ 2 ] },
  { op: 'remove', path: [ 3 ] },
  { op: 'remove', path: [ 4 ] },
  { op: 'replace', path: [ 0 ], value: 5 } ]
> diffApply(a, c)
Error: expected to find property 3 in object 1,3,5
    at diffApply (/.../node_modules/just-diff-apply/index.js:90:15)
```

Is this behaviour expected?

**Comment by angus-c** [comment]
Good catch, this is a significant issue that I overlooked.
I should either 
a) reverse sort remove ops by key
or
b) instead of deleting replace with a placeholder and finally purge placeholders

**Comment by kbhokray** [comment]
Caught the bug just before we were going prod :sweat_smile:. Reverse sort looks like a lot of work since it should be a path dependent reversal. We finally went ahead with [this](https://github.com/mirek/node-rus-diff) library which uses replace, which might be a better way IMHO.

**Comment by angus-c** [comment]
If you're still interested it's fixed in just-diff@3.0.1, I ended up sorting removes (by full path) in the just-diff app.

Thanks again for reporting this!
